File: algorithms/java/Jun.2020.challenge/Challenge_Validate_IP_Address.java
import java.util.regex.Pattern;

class Solution {
    public String validIPAddress(String IP) {
        String ipv4Pat = "^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])$";
        String ipv6Pat = "^([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}$";
        
        if(Pattern.matches(ipv4Pat, IP))
            return "IPv4";
        else if(Pattern.matches(ipv6Pat, IP))
            return "IPv6";
        
        return "Neither";
    }
}

File: algorithms/python/Jun.2020.challenge/Challenge_Validate_IP_Address.py
import re

class Solution:
    def validIPAddress(self, IP: str) -> str:
        ipv4Pat = "^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])$"
        if re.compile(ipv4Pat).match(IP):
            return "IPv4"
        
        ipv6Pat = "^([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}$"
        if re.compile(ipv6Pat).match(IP):
            return "IPv6"
        
        return "Neither"
